urlaf656038a0_Salihan04_MOBLIMA_tgz-pJ8-com_moblima_util_dataTestJ8/src/com/moblima/dataaccess/SerializeDB.java

```diff
@@ -0,0 +1,314 @@
+package com.moblima.dataaccess;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.moblima.model.Booking;
+import com.moblima.model.Cinema;
+import com.moblima.model.Cineplex;
+import com.moblima.model.Movie;
+import com.moblima.model.MovieGoer;
+import com.moblima.model.MovieTicket;
+import com.moblima.model.PublicHoliday;
+import com.moblima.model.Seat;
+import com.moblima.model.ShowTime;
+import com.moblima.model.Staff;
+import com.moblima.model.Transaction;
+import com.moblima.model.TicketPrice;
+
+public class SerializeDB {
+	
+	private List<Cineplex> cineplexes;
+	private List<Cinema> cinemas;
+	private List<Seat> seats;
+	private List<Movie> movies;
+	private List<ShowTime> showTimes;
+	private List<MovieGoer> movieGoers;
+	private List<MovieTicket> movieTickets;
+	private List<Transaction> transactions;
+	private List<Booking> bookings;
+	private List<PublicHoliday> publicHolidays;
+	private List<TicketPrice> ticketPrices;
+	private List<Staff> staffs;
+	
+	private Integer cineplexID;
+	private Integer seatID;
+	private Integer movieID;
+	private Integer showTimeID;
+	private Integer movieGoerID;
+	private Integer movieTicketID;
+	private Integer bookingID;
+	private Integer publicHolidayID;
+	private Integer staffID;
+	
+	private static String fileName = "serializeDB.dat";
+	private static SerializeDB instance;
+	
+	public static SerializeDB getInstance() {
+		if(instance == null) {
+			instance = new SerializeDB();
+		}
+		
+		return instance;
+	}
+	
+	public void saveData() {
+		try {
+			FileOutputStream fout = new FileOutputStream(fileName);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fout*. The resource is referenced at lines 64, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 79, 80, 81, 82, 83, 84, 85, 86, 87. The resource is closed at line 89. There are other execution paths that do not contain closure statements, for example, when ObjectOutputStream.writeObject throws an exception. Ensure *fout* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).